### PR TITLE
BUGFIX: SD-313 SD-318 Column/Bar chart missed some values

### PR DESCRIPTION
**Pull request by naibinh**
## PR checklist
- [x] AD/KD test cafe by using dev release
- [x] Unit tests are updated and passed
- [x] Storbybooks are passed
- [x] Squash commits
---
## Related PRs
AD: https://github.com/gooddata/gdc-analytical-designer/pull/2060
KD: https://github.com/gooddata/gdc-dashboards/pull/1673

### CODE REVIEW

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3208/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2548/ : SUCCESS in 6m 59s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1335/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3209/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/797/ : UNSTABLE in 24m 34s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3211/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2549/ : SUCCESS in 7m 14s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1337/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3212/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/799/ : SUCCESS in 21m 49s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3233/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2564/ : SUCCESS in 11m 26s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3234/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2565/ : SUCCESS in 5m 35s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1347/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3235/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/808/ : UNSTABLE in 24m 46s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3236/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2566/ : SUCCESS in 2m 04s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1348/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3237/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/809/ : SUCCESS in 21m 05s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3257/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2587/ : SUCCESS in 2m 18s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1358/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3258/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/814/ : UNSTABLE in 26m 16s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3260/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2589/ : SUCCESS in 2m 53s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3261/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2590/ : SUCCESS in 7m 52s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1359/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3262/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/815/ : UNSTABLE in 20m 11s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1360/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3263/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/816/ : SUCCESS in 18m 17s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3270/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2597/ : SUCCESS in 6m 39s

File: src/components/visualizations/chart/chartOptionsBuilder.ts [hunk 1]
```diff
@@ -1117,16 +1111,47 @@ function getCategories(
...
+ *          categories: [C2]
+ *       }]
+ * @param viewByTwoAttributes
+ */
+export function getCategoriesForTwoAttributes(viewByTwoAttributes: IViewByTwoAttributes): Array<{
+    name: string,
+    categories: string[]
+}> {
+    const { parent = [], children = [] } = viewByTwoAttributes;
+    const combinedResult = zip(parent, children).reduce(
```
Comment (nadtan) [inline src/components/visualizations/chart/chartOptionsBuilder.ts]: I don't think its necessary to use `zip` there.
you can simply use the third parameter in `reduce` to get the mapping item in `children` array.

Comment (naibinh) [reply src/components/visualizations/chart/chartOptionsBuilder.ts]: Fixed :)

**Comment by nadtan** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3294/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2615/ : SUCCESS in 11m 56s

File: stories/test_data/chart_with_2_attributes_and_null_datapoints_request.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    "execution": {
+        "afm": {
+            "measures": [{
+                "localIdentifier": "ah1EuQxwaCqs",
+                "definition": {"measure": {"item": {"identifier": "ah1EuQxwaCqs"}}}
+            }],
+            "attributes": [{
+                "displayForm": {"identifier": "label.product.id.name"},
+                "localIdentifier": "va_13"
+            }, {"displayForm": {"identifier": "label.stage.name.stagename"}, "localIdentifier": "va_17"}],
```
Comment (xMort) [inline stories/test_data/chart_with_2_attributes_and_null_datapoints_request.json]: Please fix the format. One attribute is properly formatted, the other is not. Whole JSON file format seems to be inconsistent when I'm looking at it.

Comment (naibinh) [reply stories/test_data/chart_with_2_attributes_and_null_datapoints_request.json]: it's done :)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3324/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2641/ : SUCCESS in 3m 04s

File: stories/test_data/chart_with_2_attributes_and_null_datapoints_request.json (hunk 2 above)
Comment (xMort) [reply stories/test_data/chart_with_2_attributes_and_null_datapoints_request.json]: It looks way better now. Please squash the commits and you can merge the pull request.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3337/ : SUCCESS in 30s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2650/ : SUCCESS in 3m 40s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/706/ : SUCCESS in 45s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2651/ : SUCCESS in 5m 00s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1391/ : SUCCESS in 28m 17s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/689/ : SUCCESS in 25m 40s

**Pull request merged**